blocks/bricks/sequence_generators.py

```diff
@@ -436,6 +436,9 @@ def emit(self, readouts):
 
     @application
     def cost(self, readouts, outputs):
+        # WARNING: unfortunately this application method works
+        # just fine when `readouts` and `outputs` have
+        # different dimensions. Be careful!
         probs = self.probs(readouts)
```

Comment:
I guess it would be useful to have error checking nodes in theano or blocks for cases like this where you want to raise an error during execution rather than compilation time without writing something cryptic that uses a computation that breaks on purpose when the shapes aren't the same. Does something like this already exist?